lib/wrappers/openssl.nim

```diff
@@ -191,16 +206,46 @@ const
 
 proc TLSv1_method*(): PSSL_METHOD{.cdecl, dynlib: DLLSSLName, importc.}
 
+# TLS_method(), TLS_server_method(), TLS_client_method() are introduced in 1.1.0
+# and support SSLv3, TLSv1, TLSv1.1 and TLSv1.2
+# SSLv23_method(), SSLv23_server_method(), SSLv23_client_method() are removed in 1.1.0
+
 when compileOption("dynlibOverride", "ssl"):
-  proc SSL_library_init*(): cint {.cdecl, dynlib: DLLSSLName, importc, discardable.}
-  proc SSL_load_error_strings*() {.cdecl, dynlib: DLLSSLName, importc.}
-  proc SSLv23_client_method*(): PSSL_METHOD {.cdecl, dynlib: DLLSSLName, importc.}
+  # Static linking
+  when defined(openssl_before_1_1):
```

Comment:
I looked into the OpenSSL library initialisation and found a nice way to support both without the need for our own pesky defines: https://wiki.openssl.org/index.php/Library_Initialization#libssl_Initialization

Perhaps we could perform the same check against ``OPENSSL_VERSION_NUMBER`` statically? This might require some clever C code emitting but it would be nice.

The static ``getOpenSSLVersion`` could also use this constant. Also, from what I've seen, the docs seem to imply that ``OpenSSL_version_num`` is available in much earlier versions than 1.1.0.

Reply:
The define is meant to provide a simple way to change behavior in other modules without breaking the builds when some procs are undefined. My first use case would be disabling SSL cert checks.